### ISSUE TITLE
FUSESMB will not install via HaikuDepot

### ISSUE THREAD

**Issue by tthoms** [haikuports/haikuports#4728]
Upon pressing "install", the installer reports that "nothing provides lib:libreadline>=7.0 neede by samba3.6.25-6".  This despite the presence of libreadline 7 and 8 on the computer.

Dell Latitude D610 running Haiku x 86  beta1 hrev52295+132

**Comment by diversys** [comment]
Can't reproduce on a recent nightly. What happens if you run `pkgman full-sync`?

**Comment by waddlesplash** [comment]
No reply.

**Comment by Angsereg** [comment]
I get this same "Package problems" on a fresh install of nightly hrev 54722 (x86_gcc2h). This is on an HP t5740e thin client, running nightly hrev 54722 x86_gcc2h, sporting a 4GB SSD on the IDE bus, and a single stick of 4GB DDR3, and an Intel Atom N280.

Should I try the `pkgman full-sync`? Is there any relevant info I can provide that will make it easier to narrow down the underlying cause?

**Comment by diversys** [comment]
The problem is that there is no lib:libreadline>=7.0 on x86_gcc2

**Comment by Angsereg** [comment]
Then, is the samba package (and consequently the FuseSMB package) a 64-bit-only package, then?

There is a "readline" package listed as "Active" in HaikuDepot, as well as readline6_x86, readline7_x86, and readline_x86 packages that are listed as "Available". The "readline" and "readline_x86" packages show the same version numbers: 8.0.4-1. 

Do none of those packages "provide" lib:libreadline, even though "readline" does physically provide the lib/libreadline.so.8{,.0} and lib/libhistory.so.8{,.0} files & symlinks?

**Comment by diversys** [comment]
Samba package used to work on x86_gcc2 at the time when libreadline 6 was the default and broke when we bumped it to libreadline 7.
I guess we can either depend on libreadline 6 in samba [recipe](https://github.com/haikuports/haikuports/blob/master/net-fs/samba/samba-3.6.25.recipe#L69) (I'm not sure if that's wanted tho) or try to fix readline 7 (or even readline 8?) to build using gcc2.

**Comment by diversys** [comment]
Actually, the problem seems to be entirely different, so disregard my previous message for now. Here's where samba build fails on x86_gcc2:
```
Compiling lib/netapi/netapi.c
In file included from /boot/system/develop/headers/os/support/SupportDefs.h:15,
                 from /boot/system/develop/headers/os/support/ByteOrder.h:11,
                 from /boot/system/develop/headers/bsd/endian.h:16,
                 from /boot/system/develop/headers/posix/netinet/in.h:10,
                 from /sources/samba-3.6.25/lib/replace/system/network.h:43,
                 from /sources/samba-3.6.25/lib/util/util_net.h:24,
                 from /sources/samba-3.6.25/source3/include/includes.h:507,
                 from /sources/samba-3.6.25/source3/lib/netapi/netapi.c:20:
/boot/system/develop/headers/posix/inttypes.h:32: warning: `PRId32' redefined
/sources/samba-3.6.25/source3/../lib/popt/../replace/replace.h:75: warning: this is the location of the previous definition
/boot/system/develop/headers/posix/inttypes.h:35: warning: `PRIi32' redefined
/sources/samba-3.6.25/source3/../lib/popt/../replace/replace.h:88: warning: this is the location of the previous definition
/boot/system/develop/headers/posix/inttypes.h:39: warning: `PRId64' redefined
/sources/samba-3.6.25/source3/../lib/popt/../replace/replace.h:78: warning: this is the location of the previous definition
/boot/system/develop/headers/posix/inttypes.h:42: warning: `PRIi64' redefined
/sources/samba-3.6.25/source3/../lib/popt/../replace/replace.h:91: warning: this is the location of the previous definition
/boot/system/develop/headers/posix/inttypes.h:78: warning: `PRIu32' redefined
/sources/samba-3.6.25/source3/../lib/popt/../replace/replace.h:101: warning: this is the location of the previous definition
/boot/system/develop/headers/posix/inttypes.h:91: warning: `PRIu64' redefined
/sources/samba-3.6.25/source3/../lib/popt/../replace/replace.h:104: warning: this is the location of the previous definition
In file included from /boot/system/develop/headers/os/support/ByteOrder.h:11,
                 from /boot/system/develop/headers/bsd/endian.h:16,
                 from /boot/system/develop/headers/posix/netinet/in.h:10,
                 from /sources/samba-3.6.25/lib/replace/system/network.h:43,
                 from /sources/samba-3.6.25/lib/util/util_net.h:24,
                 from /sources/samba-3.6.25/source3/include/includes.h:507,
                 from /sources/samba-3.6.25/source3/lib/netapi/netapi.c:20:
/boot/system/develop/headers/os/support/SupportDefs.h:21: warning: redefinition of `uint8_t'
/boot/system/develop/headers/posix/stdint.h:14: warning: `uint8_t' previously declared here
/boot/system/develop/headers/os/support/SupportDefs.h:22: warning: redefinition of `int16_t'
/boot/system/develop/headers/posix/stdint.h:16: warning: `int16_t' previously declared here
/boot/system/develop/headers/os/support/SupportDefs.h:23: warning: redefinition of `uint16_t'
/boot/system/develop/headers/posix/stdint.h:17: warning: `uint16_t' previously declared here
/boot/system/develop/headers/os/support/SupportDefs.h:24: conflicting types for `int32_t'
/boot/system/develop/headers/posix/stdint.h:19: previous declaration of `int32_t'
/boot/system/develop/headers/os/support/SupportDefs.h:25: conflicting types for `uint32_t'
/boot/system/develop/headers/posix/stdint.h:20: previous declaration of `uint32_t'
/boot/system/develop/headers/os/support/SupportDefs.h:26: warning: redefinition of `int64_t'
/boot/system/develop/headers/posix/stdint.h:22: warning: `int64_t' previously declared here
/boot/system/develop/headers/os/support/SupportDefs.h:27: warning: redefinition of `uint64_t'
/boot/system/develop/headers/posix/stdint.h:23: warning: `uint64_t' previously declared here
The following command failed:
gcc -D_BSD_SOURCE -I. -I/sources/samba-3.6.25/source3 -I/sources/samba-3.6.25/source3/../lib/popt -I/sources/samba-3.6.25/source3/../lib/iniparser/src -Iinclude -I./include  -I. -I. -I./../lib/replace -I./librpc -I./.. -I./../lib/talloc -I./../lib/tevent -I../lib/tdb/include -DHAVE_CONFIG_H  -Iinclude -I./include -I. -I. -I./../lib/replace -I./librpc -I./.. -I./../lib/popt  -I/sources/samba-3.6.25/source3/lib -I.. -D_SAMBA_BUILD_=3 -D_SAMBA_BUILD_=3 -fPIC -c lib/netapi/netapi.c -o lib/netapi/netapi.o
Makefile:1674: recipe for target 'lib/netapi/netapi.o' failed
make: *** [lib/netapi/netapi.o] Error 1
make: *** Waiting for unfinished jobs....
Warning: Command '['bash', '-c', '. /wrapper-script']' returned non-zero exit status 2.
Error: Build has failed - stopping.
```

**Comment by Angsereg** [comment]
Well, I guess that explains it - the dependencies need to build first. I'll hold off on trying to install this until samba's able to build.

**Comment by diversys** [comment]
Fixed in 8a531cfdddfc7ba716c5d482a6f6cac48103f633.
